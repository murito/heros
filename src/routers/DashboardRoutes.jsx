import React from 'react'
import { Navbar } from '../components/ui/Navbar'
import { Switch, Route, Redirect } from 'react-router-dom';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { HeroScreen } from '../components/heros/HeroScreen';
import { DcScreen } from '../components/dc/DcScreen';
import { SearchScreen } from '../components/search/SearchScreen';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />

            <div className="container-fluid mt-5 px-5">
                <Switch>
                    <Route exact path="/dc" component={ DcScreen }/>
                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroeId" component={ HeroScreen }/>
                    <Route exact path="/search" component={ SearchScreen }/>

                    <Redirect to="/marvel" />
                </Switch>    
            </div>   
        </>
    )
}
