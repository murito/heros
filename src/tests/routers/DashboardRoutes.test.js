import { mount } from 'enzyme'
import { MemoryRouter } from 'react-router';
import { AuthContext } from '../../auth/AuthContext';
import { DashboardRoutes } from '../../routers/DashboardRoutes'

describe('Tests on <DashboardRoutes />', () => {
    const contextValue = {
        user: {
            logged: true,
            name: 'Juanito'
        },
        dispatch: jest.fn()
    };

    test('should be shown itself in the right way', () => {

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        );
    
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim() ).toBe( 'Juanito' );
    })
})
