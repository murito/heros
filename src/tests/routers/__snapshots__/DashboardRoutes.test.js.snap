// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <DashboardRoutes /> should be shown itself in the right way 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          <span
            className="search"
          >
            <i
              className="fas fa-search"
            />
             Search
          </span>
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Juanito
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          <span
            className="search"
          >
            Logout
          </span>
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container-fluid mt-5 px-5"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="hero-list animate__animated animate__fadeIn"
      >
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Spider Man"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Peter Parker
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Spider Man
            </h5>
            <p
              className="card-text"
            >
              Peter Parker
            </p>
            <small
              className="text-muted"
            >
              Amazing Fantasy #15
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Captain America"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Steve Rogers
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Captain America
            </h5>
            <p
              className="card-text"
            >
              Steve Rogers
            </p>
            <small
              className="text-muted"
            >
              Captain America Comics #1
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Iron Man"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Tony Stark
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Iron Man
            </h5>
            <p
              className="card-text"
            >
              Tony Stark
            </p>
            <small
              className="text-muted"
            >
              Tales of Suspense #39
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Thor"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Thor Odinson
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Thor
            </h5>
            <p
              className="card-text"
            >
              Thor Odinson
            </p>
            <small
              className="text-muted"
            >
              Journey into Myster #83
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Hulk"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Bruce Banner
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Hulk
            </h5>
            <p
              className="card-text"
            >
              Bruce Banner
            </p>
            <small
              className="text-muted"
            >
              The Incredible Hulk #1
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Wolverine"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  James Howlett
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Wolverine
            </h5>
            <p
              className="card-text"
            >
              James Howlett
            </p>
            <small
              className="text-muted"
            >
              The Incredible Hulk #180
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Daredevil"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Matthew Michael Murdock
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Daredevil
            </h5>
            <p
              className="card-text"
            >
              Matthew Michael Murdock
            </p>
            <small
              className="text-muted"
            >
              Daredevil #1
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Hawkeye"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Clinton Francis Barton
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Hawkeye
            </h5>
            <p
              className="card-text"
            >
              Clinton Francis Barton
            </p>
            <small
              className="text-muted"
            >
              Tales of Suspense #57
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Cyclops"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Scott Summers
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Cyclops
            </h5>
            <p
              className="card-text"
            >
              Scott Summers
            </p>
            <small
              className="text-muted"
            >
              X-Men #1
            </small>
          </div>
        </div>
        <div
          className="card ms-3 hero-card"
        >
          <img
            alt="Silver Surfer"
            className="card-img"
            src=""
          />
          <div
            className="card-body"
          >
            <div
              className="characters marvel"
            >
              <a
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                <h4
                  className="text-center"
                >
                  Norrin Radd
                </h4>
              </a>
            </div>
            <h5
              className="card-title"
            >
              Silver Surfer
            </h5>
            <p
              className="card-text"
            >
              Norrin Radd
            </p>
            <small
              className="text-muted"
            >
              The Fantastic Four #48
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
